spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java

```diff
@@ -146,7 +169,7 @@ protected InputFile getInputFile(String location) {
     return inputFiles.get(location);
   }
 
-  protected Map<Integer, ?> constantsMap(FileScanTask task, Schema readSchema) {
+  protected Map<Integer, ?> constantsMap(CST task, Schema readSchema) {
```

Comment:
If we change the boundary to `ScanTask`, then this method can accept `ContentScanTask<?>`.